spark_expectations/utils/actions.py

```diff
@@ -94,6 +97,338 @@ def create_rules_map(_rule_map: Dict[str, str]) -> Any:
+                                            _context.spark.sql(
+                                                _dq_rule["expectation" + "_" + _key]
+                                            )
+                                            .toJSON()
+                                            .collect(),
+                                        )
+                                    ]
+                                ),
+                                _context.get_run_date,
+                            )
+                        )
+                    print("_querydq_output in agg_query_dq_detailed:", querydq_output)
+
+                if SparkExpectationsActions.match_parentheses(_dq_rule["expectation"]):
+                    # pattern = r"(\(.*\)) ([<>!=]=?) ((\d+)|(\(.*\)))|(\(.*\))"
+                    pattern = r"(\(.*\))\s*([<>!=]=?)\s*((\d+)|(\(.*\)))|(\(.*\))"
+                    match = re.search(pattern, _dq_rule["expectation"])
+                    if match:
+                        print("Match found")
+                        print("Group 1:", match.group(1))
+                        print("Group 2:", match.group(2))
+                        print("Group 3:", match.group(3))
+                        print("Group 4:", match.group(4))  # digits
+                        print("Group 5:", match.group(5))
+
+                        # function to execute SQL and get the result
+                        def execute_sql_and_get_result(
+                            _se_context: SparkExpectationsContext, query: str
+                        ) -> int:
+                            return (
+                                _se_context.spark.sql(
+                                    f"SELECT ({query}) AS OUTPUT"
+                                ).collect()[0][0]
+                                if query
+                                else None
+                            )
+                            # if query:
+                            #     return context.spark.sql(f"SELECT ({query}) AS OUTPUT").collect()[0][0]
+                            # return None
+
+                        # function to get the query outputs
+                        _querydq_source_query_output = execute_sql_and_get_result(
+                            _context, match.group(1)
+                        )
+                        _querydq_target_query_output = match.group(
+                            4
+                        ) or execute_sql_and_get_result(_context, match.group(5))
+
+                        # assignment of actual_outcome and expected_outcome
+                        actual_outcome = _querydq_source_query_output
+                        expected_outcome = (
+                            str(match.group(2)) + str(_querydq_target_query_output)
+                            if _querydq_target_query_output
+                            else None
+                        )
+
+                        # if match.group(1):
```

Comment:
Can we clean up these commented code please

Reply:
Done